svc-bip-api/src/test/java/gov/va/vro/bip/service/BipApiServiceThroughRMQTest.java

```diff
@@ -0,0 +1,175 @@
+    String baseUrl = HTTPS + CLAIM_URL;
+    String claimUrl = baseUrl + String.format(CLAIM_DETAILS, GOOD_CLAIM_ID);
+    String badClaimUrl = baseUrl + String.format(CLAIM_DETAILS, BAD_CLAIM_ID);
+
+    Mockito.doReturn(resp200)
+        .when(restTemplate)
+        .exchange(
+            ArgumentMatchers.eq(claimUrl),
+            ArgumentMatchers.eq(HttpMethod.GET),
+            ArgumentMatchers.any(HttpEntity.class),
+            ArgumentMatchers.eq(String.class));
+    Mockito.doReturn(resp404)
+        .when(restTemplate)
+        .exchange(
+            ArgumentMatchers.eq(badClaimUrl),
+            ArgumentMatchers.eq(HttpMethod.GET),
+            ArgumentMatchers.any(HttpEntity.class),
+            ArgumentMatchers.eq(String.class));
+    mockBipApiProp();
+
+    BipClaim result = (BipClaim) rabbitTemplate.convertSendAndReceive(
+              exchangeName,
+              "getClaimDetailsQueue",
+              GOOD_CLAIM_ID);
+    assertEquals(500, result.statusCode);
+
+
+    result = (BipClaim) rabbitTemplate.convertSendAndReceive(
+            exchangeName,
+            "getClaimDetailsQueue",
+            BAD_CLAIM_ID);
+    assertEquals(500, result.statusCode);
+  }
+
+  @Test
+  public void testSetClaimToRfdStatus() {
+    String baseUrl = HTTPS + CLAIM_URL;
+    String goodUrl = baseUrl + String.format(UPDATE_CLAIM_STATUS, GOOD_CLAIM_ID);
+    String badUrl = baseUrl + String.format(UPDATE_CLAIM_STATUS, BAD_CLAIM_ID);
+    ResponseEntity<String> resp200 = ResponseEntity.ok("{}");
+    ResponseEntity<String> resp500 =
+        ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("error");
+    Mockito.doReturn(resp200)
+        .when(restTemplate)
+        .exchange(
+            ArgumentMatchers.eq(goodUrl),
+            ArgumentMatchers.eq(HttpMethod.PUT),
+            ArgumentMatchers.any(HttpEntity.class),
+            ArgumentMatchers.eq(String.class));
+    Mockito.doReturn(resp500)
+        .when(restTemplate)
+        .exchange(
+            ArgumentMatchers.eq(badUrl),
+            ArgumentMatchers.eq(HttpMethod.PUT),
+            ArgumentMatchers.any(HttpEntity.class),
+            ArgumentMatchers.eq(String.class));
+    mockBipApiProp();
+
+    //positive test
+    BipUpdateClaimResp result = (BipUpdateClaimResp) rabbitTemplate.convertSendAndReceive(
+            exchangeName,
+            "setClaimToRfdStatusQueue",
+            GOOD_CLAIM_ID);
+    assertEquals(500, result.statusCode);
+
+    //negative test
+    result = (BipUpdateClaimResp) rabbitTemplate.convertSendAndReceive(
+            exchangeName,
+            "setClaimToRfdStatusQueue",
+            BAD_CLAIM_ID);
+    assertEquals(500, result.statusCode);
+  }
+  @BeforeEach private void setUp() {rabbitAdmin.purgeQueue(queueName, true);}
+  @AfterEach private void tearDown() {rabbitAdmin.purgeQueue(queueName, true);}
+
+  static  String queueName = null;
```

Comment:
Good point.  I don't know how this test ever worked.  I'm guessing rabbitAdmin just guesses que name if it's not provided instead of throwing.  But you're right this is bad.

Reply:
fixed